src/Types/VoidType.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace ILIAS\Types;
+
+
+class VoidType implements Type {
+
+	protected static $instance = null;
+
+	/**
+	 * Just to make it protected.
+	 * VoidValue constructor.
+	 */
+	protected function __construct() {
+	}
+
+	/**
+	 * @return VoidType
+	 */
+	public static function instance(){
```

Comment:
A void type seems to be one of the few valid cases for a singleton. What's the advantage of a factory? Multiple instances of a void type? A global void factory? Sounds like overengineering to me.